### DISCUSSION TITLE
SWIG build trouble with PCRE2

### DISCUSSION THREAD

**Discussion by scottcarnold** [swig/swig#2497] [General]
I'm attempting to use SWIG on RHEL8.  Neither SWIG nor PCRE2 were available through the dnf package manager, so I'm building both from source.  I have pcre2-10.42 and swig-4.1.1.  PCRE2 built okay with defaults (./configure; make; make install).  The pcre2-config executable installed to /usr/local/bin and the libpcre2 stuff went to /usr/local/lib and a couple of header files to /usr/local/include.

If I then do a ./configure on swig, it gives error Cannot find pcre2-config script from PCRE2.  So presumably it's not looking for pcre2 in the location the pcre2 build system installs to.  So I then tried setting the --with-pcre2-prefix and --with-pcre2-exec-prefix flags like so:

`./configure --with-pcre2-prefix=/usr/local --with-pcre2-exec-prefix=/usr/local/bin`  

When I run it this way the configure works but make results in an error.  I'm not entirely sure what --with-pcre2-prefix is wanting exactly; I tried /usr/local, /usr/local/include, and usr/local/lib, all which result in the same error.  The error is:

`Swig/misc.o: In function 'Swig_string_regex':
/opt/swig-4.1.1/Source/Swig/misc.c:1463: undefined reference to 'pcre2_compile_8'
/opt/swig-4.1.1/Source/Swig/misc.c:1471: undefined reference to 'pcre2_match_data_create_from_pattern_8'
...`

There are multiple undefined references; I'm just including the first 2 here.  Any suggestions on what is going wrong here?

Thank you.

**Comment by scottcarnold** [comment]
Minor update:  It worked to use the Tools/pcre-build.sh script (after copying the PCRE2 tarball into the SWIG folder) and then do the ./configure.  So this was my workaround.  But I am still interested in knowing why I was not able to install PCRE2 as shown and get the SWIG build to work from it.

**Comment by wsfulton** [comment]
Looks like a linker problem. Looking at [configure.ac](https://github.com/swig/swig/blob/92b3c8a54fe9346f40a24494842f0e412ad62a99/configure.ac#L67) file, try define `PCRE2_LIBS` for the linker to fix (something like `-lpcre2-8`).
